pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -247,16 +216,18 @@ func (c *Controller) syncOwnedProvisionRequest(ctx context.Context, wl *kueue.Wo
 		attempt := int32(1)
 		shouldCreatePr := false
 		if exists {
-			if isFailed(oldPr) || (isBookingExpired(oldPr) && !workload.IsAdmitted(wl)) {
+			if (isFailed(oldPr) || (isBookingExpired(oldPr) && !workload.IsAdmitted(wl))) &&
 				// if the workload is Admitted we don't want to retry on BookingExpired
+				ac != nil && ac.State == kueue.CheckStatePending {
+				// if the workload is in Retry/Rejected state we don't create another ProvReq
 				attempt = getAttempt(log, oldPr, wl.Name, checkName)
-				if attempt <= c.maxRetries {
-					remainingTime := c.remainingTimeToRetry(oldPr, attempt)
-					if remainingTime <= 0 {
+				if attempt <= ptr.Deref(retryStrategy.BackoffLimitCount, kueue.DefaultBackoffLimitCount) {
+					remainingTimeToRetry := c.remainingTimeToRetry(wlInfo, oldPr, attempt, prc)
```

Comment:
```suggestion
					remainingTime := c.remainingTimeToRetry(wlInfo, oldPr, attempt, prc)
```
nit, your name makes sens, but this is a delicate PR, so I would prefer to minimize review distractions by renames. We can follow up with a cleanup PR. Same for requeueAfter.

Reply:
Done